src/transxchange/read.rs

```diff
@@ -101,16 +102,45 @@ fn update_validity_period_from_transxchange(
     CollectionWithId::new(datasets)
 }
 
-fn get_operator(root: &Element) -> Result<&Element> {
-    root.try_only_child("Operators")?.try_only_child("Operator")
+fn get_operator<'a>(transxchange: &'a Element, operator_ref: &String) -> Result<&'a Element> {
+    let is_operator_ref = |operator: &&Element| {
+        operator
+            .attr("id")
+            .map(|id| id == operator_ref)
+            .unwrap_or(false)
```

Comment:
```rust
operator
            .attr("id")
            .filter(|id| id == &operator_ref)
            .is_some()
```

Reply:
Done in 28c8cffa0c21ba886ef49e5b45af5b32ee5677bc.